### PR TITLE
feat(CollectiblesView): Add combobox/popup with custom filtering options

### PR DESCRIPTION
**Pull request by caybro**
.. to filter by community or collection name

- make the HistoryView own filter button look like the other combos
- fix some cosmetic issues for StatusCombo in small/secondary mode
- fix StatusBaseInput bg color in dark mode (was invisible)
- update CollectiblesViewPage with options to include regular and/or
community collectibles

Fixes https://github.com/status-im/status-desktop/issues/12969
Fixes https://github.com/status-im/status-desktop/issues/12948

### What does the PR do

Adds a filtering combo to the collectibles view

### Affected areas

Wallet/CollectiblesView

### Screenshot of functionality (including design for comparison)

- [x] I've checked the design and this PR matches it

Snímek obrazovky z 2023-12-13 16-00-11

Snímek obrazovky z 2023-12-13 16-00-55

Storybook with optional collectibles' sources:
image

### CODE REVIEW

File: ui/app/AppLayouts/Wallet/controls/FilterComboBox.qml [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+
+import utils 1.0
+import shared.controls 1.0
+
+import SortFilterProxyModel 0.2
+
+ComboBox {
+    id: root
+
+    required property var controller
```
Comment (caybro) [inline ui/app/AppLayouts/Wallet/controls/FilterComboBox.qml:21]: This should probably be just the 2 models instead...

File: ui/app/AppLayouts/Wallet/controls/FilterComboBox.qml
Comment (noeliaSD) [inline ui/app/AppLayouts/Wallet/controls/FilterComboBox.qml:1]: Where is it the `storybook` page?? :wink:

File: ui/app/AppLayouts/Wallet/controls/FilterComboBox.qml (hunk 1 above)
Comment (caybro) [reply ui/app/AppLayouts/Wallet/controls/FilterComboBox.qml]: Fixed

File: ui/app/AppLayouts/Wallet/controls/FilterComboBox.qml [hunk 2]
```diff
@@ -0,0 +1,319 @@
...
+
+import utils 1.0
+import shared.controls 1.0
+
+import SortFilterProxyModel 0.2
+
+ComboBox {
+    id: root
+
+    required property var regularTokensModel
```
Comment (noeliaSD) [inline ui/app/AppLayouts/Wallet/controls/FilterComboBox.qml:21]: Will these be the ones that are NOT part of any collection? Maybe a comment could help to clarify!

File: ui/app/AppLayouts/Wallet/controls/FilterComboBox.qml [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+    function clearFilter() {
+        d.selectedFilterGroupIds = []
+    }
+
+    enabled: d.combinedProxyModel.count || d.uncategorizedModel.count
+    opacity: enabled ? 1 : 0.3
+
+    displayText: qsTr("Collection")
+
+    horizontalPadding: 12
```
Comment (noeliaSD) [inline ui/app/AppLayouts/Wallet/controls/FilterComboBox.qml:37]: Some `Style.current.value`?!

**Review by noeliaSD — CHANGES_REQUESTED** [review]
From the images shared in the PR it looks nice!! But I could not test the new combobox added since there's no `storybook` page for it (it'd be nice to add it ;) ) to test isolated and I experience a crash when trying to open `Wallet main layout`. More info in the following comments:

- There's the following line that probably should not be present when regular collectibles model is empty:
Screenshot 2023-12-19 at 10 22 47

- Whenever I enter in `Wallet Main View` there's the following crash (tried on master and it's not reproducible):
```
DBG 2023-12-19 10:39:52.573+01:00 NewBE_callPrivateRPC                       topics="rpc" tid=144102 file=core.nim:27 rpc_method=wakuext_slowdownArchivesImport
DBG 2023-12-19 10:39:52.731+01:00 NewBE_callPrivateRPC                       topics="rpc" tid=144102 file=core.nim:27 rpc_method=accounts_hasPairedDevices
DBG 2023-12-19 10:39:52.765+01:00 NewBE_callPrivateRPC                       topics="rpc" tid=144102 file=core.nim:27 rpc_method=wallet_getOwnedCollectiblesAsync
Warning: void ManageTokensController::loadSettings() Missing symbol while reading tokens settings (:0, )
Warning: qrc:/app/AppLayouts/Wallet/views/AssetsDetailView.qml:32:9: Unable to assign [undefined] to QString (qrc:/app/AppLayouts/Wallet/views/AssetsDetailView.qml:32, )
Traceback (most recent call last)
/status-desktop/src/nim_status_client.nim(208) nim_status_client
/status-desktop/src/nim_status_client.nim(205) mainProc
/status-desktop/vendor/nimqml/src/nimqml/private/qguiapplication.nim(29) exec
/status-desktop/vendor/nimqml/src/nimqml/private/qobject.nim(52) qobjectCallback
/status-desktop/vendor/nimqml/src/nimqml/private/qobject.nim(37) onSlotCalled
/status-desktop/vendor/nimbus-build-system/vendor/Nim/lib/core/macros.nim(558) onSlotCalled
/status-desktop/src/app/core/signals/signals_manager.nim(63) receiveSignal
/status-desktop/src/app/core/signals/signals_manager.nim(60) processSignal
/status-desktop/src/app/core/eventemitter.nim(58) emit
/status-desktop/src/app/modules/shared_modules/collectibles/events_handler.nim(116) :anonymous
/status-desktop/src/app/modules/shared_modules/collectibles/events_handler.nim(69) handleApiEvents
/status-desktop/src/app/modules/shared_modules/collectibles/controller.nim(156) :anonymous
/status-desktop/src/app/modules/shared_modules/collectibles/controller.nim(126) processGetOwnedCollectiblesResponse
/status-desktop/src/app/modules/shared_models/collectibles_model.nim(94) setIsFetching
/status-desktop/src/app/modules/shared_models/collectibles_model.nim(325) checkLoadingItems
/status-desktop/src/app/modules/shared_models/collectibles_model.nim(316) removeLoadingItems
/status-desktop/vendor/nimqml/src/nimqml/private/qabstractitemmodel.nim(196) endRemoveRows
SIGSEGV: Illegal storage access. (Attempt to read from nil?)
```

File: ui/StatusQ/src/StatusQ/Controls/StatusBaseInput.qml [hunk 4]
```diff
@@ -278,7 +278,7 @@ Item {
     Rectangle {
         id: background
         anchors.fill: parent
-        color: root.showBackground ? Theme.palette.baseColor2 : "transparent"
+        color: root.showBackground ? Theme.palette.statusAppNavBar.backgroundColor : "transparent"
```
[the remark this answers is not in the document]
Comment (caybro) [reply ui/StatusQ/src/StatusQ/Controls/StatusBaseInput.qml:281]: Yeah... I couldn't really find a better suited color :/

File: ui/app/AppLayouts/Wallet/views/CollectiblesView.qml [hunk 5]
```diff
@@ -87,6 +88,16 @@ ColumnLayout {
                     d.controller.settingsDirty
                     return d.controller.filterAcceptsSymbol(model.symbol) && (customFilter.isCommunity ? !!model.communityId : !model.communityId)
                 }
+            },
+            ExpressionFilter {
```
[the remark this answers is not in the document]
Comment (caybro) [reply ui/app/AppLayouts/Wallet/views/CollectiblesView.qml:92]: Yeah, they are mutually exclusive, that's why they have the `enabled` property set for one or the other case; seemed more straightforward to me, the rest of the properties is exactly the same

File: ui/app/AppLayouts/Wallet/controls/FilterComboBox.qml (hunk 3 above)
Comment (caybro) [reply ui/app/AppLayouts/Wallet/controls/FilterComboBox.qml:37]: There's none for `12`

File: ui/app/AppLayouts/Wallet/controls/FilterComboBox.qml (hunk 2 above)
Comment (caybro) [reply ui/app/AppLayouts/Wallet/controls/FilterComboBox.qml:21]: Yup, will add comments to what is what

File: ui/app/AppLayouts/Wallet/controls/FilterComboBox.qml
Comment (caybro) [reply ui/app/AppLayouts/Wallet/controls/FilterComboBox.qml:1]: It's in CollectiblesViewPage; imho it doesn't make sense to have a standalone page for this, it needs some view with different collectibles

[the remark this answers is not in the document]
Comment (caybro) [reply ui/app/AppLayouts/Wallet/controls/FilterComboBox.qml:1]: Yeah maybe it's too generic but it's kinda hard to find a nice unique name with hundreds of QML files :) That's why it's inside the "Wallet/controls" subdir, it's not really useful outside its context

File: ui/StatusQ/src/wallet/managetokenscontroller.cpp [hunk 6]
```diff
@@ -478,6 +485,46 @@ void ManageTokensController::rebuildCommunityTokenGroupsModel()
     qCDebug(manageTokens) << "!!! GROUPS MODEL REBUILT WITH GROUPS:" << communityIds;
 }
 
+void ManageTokensController::rebuildRegularTokenGroupsModel()
+{
+    QStringList collectionIds;
+    QList<TokenData> result;
+
+    const auto count = m_regularTokensModel->count();
```
[the remark this answers is not in the document]
Comment (caybro) [reply ui/StatusQ/src/wallet/managetokenscontroller.cpp:493]: It's initialized right in the CTOR

File: ui/StatusQ/src/wallet/managetokenscontroller.cpp [hunk 7]
```diff
@@ -478,6 +485,46 @@ void ManageTokensController::rebuildCommunityTokenGroupsModel()
...
+            if (tokenGroup != result.cend()) {
+                const auto row = std::distance(result.cbegin(), tokenGroup);
+                TokenData updTokenGroup = result.takeAt(row);
+                updTokenGroup.balance = updTokenGroup.balance.toInt() + 1;
+                result.insert(row, updTokenGroup);
+            }
+        }
+    }
+
+    m_regularTokenGroupsModel->clear();
```
[the remark this answers is not in the document]
Comment (caybro) [reply ui/StatusQ/src/wallet/managetokenscontroller.cpp:521]: Same here, created in CTOR

File: ui/StatusQ/src/wallet/managetokenscontroller.cpp [hunk 8]
```diff
@@ -345,6 +349,9 @@ void ManageTokensController::parseSourceModel()
     reloadCommunityIds();
     m_communityTokensModel->setCommunityIds(m_communityIds);
 
+    // build collections
```
[the remark this answers is not in the document]
Comment (caybro) [reply ui/StatusQ/src/wallet/managetokenscontroller.cpp:352]: No, these are really collections; the other one being "community groups" (already existing code)

**Comment by caybro** [comment]
> 
> * There's the following line that probably should not be present when regular collectibles model is empty:
> 

Yeah! Will remove it, thanks

> * Whenever I enter in `Wallet Main View` there's the following crash (tried on master and it's not reproducible):
> 

Hmm, I can't really reproduce this :/ Will rebase onto latest master and try again

**Comment by noeliaSD** [comment]
After `storybook` inspection:

- Icon color / background looks different than in the dsign. Here the design:
Screenshot 2023-12-19 at 11 38 33

- I see this warning: `status-desktop/ui/app/AppLayouts/Wallet/views/CollectiblesView.qml:280:9: QML Connections: Detected function "onIsFetchingChanged" in Connections element. This is probably intended to be a signal handler but no signal of the target matches the name.`

**Comment by caybro** [comment]
> After `storybook` inspection:
> 
> * Icon color / background looks different than in the dsign. Here the design:

Uhm yeah... you're right... it's an icon that's supposed to look like a button; I'll think of something

> * I see this warning: `status-desktop/ui/app/AppLayouts/Wallet/views/CollectiblesView.qml:280:9: QML Connections: Detected function "onIsFetchingChanged" in Connections element. This is probably intended to be a signal handler but no signal of the target matches the name.`

Unrelated, it's because of the mock models in storybook that lack the signals the real models have

File: ui/app/AppLayouts/Wallet/controls/FilterComboBox.qml (hunk 2 above)
Comment (caybro) [reply ui/app/AppLayouts/Wallet/controls/FilterComboBox.qml]: Fixed

**Comment by caybro** [comment]
> > * There's the following line that probably should not be present when regular collectibles model is empty:
> 
> Yeah! Will remove it, thanks

Fixed

> > * Whenever I enter in `Wallet Main View` there's the following crash (tried on master and it's not reproducible):
> 
> Hmm, I can't really reproduce this :/ Will rebase onto latest master and try again

Rebased

**Comment by noeliaSD** [comment]
> > > * There's the following line that probably should not be present when regular collectibles model is empty:
> > 
> > 
> > Yeah! Will remove it, thanks
> 
> Fixed
> 
> > > * Whenever I enter in `Wallet Main View` there's the following crash (tried on master and it's not reproducible):
> > 
> > 
> > Hmm, I can't really reproduce this :/ Will rebase onto latest master and try again
> 
> Rebased

After rebase I still see exactly the same crash and log. @alexjba, @micieslak , @alexandraB99 could you try?

**Comment by noeliaSD** [comment]
Also another detail. The searcher placeholder should be:

Screenshot 2023-12-19 at 14 15 36

And allow the search by name or id.

**Comment by caybro** [comment]
> Also another detail. The searcher placeholder should be:
> 
> Screenshot 2023-12-19 at 14 15 36
> And allow the search by name or id. [Design](https://www.figma.com/file/eM26pyHZUeAwMLviaS1KJn/%E2%9A%99%EF%B8%8F-Wallet-Settings%3A-Manage-Tokens?type=design&node-id=71%3A103480&mode=design&t=HBhmgzvmu47bpf8o-1)

Right, I'll extend the search to include the collection ID as well

**Comment by caybro** [comment]
NB: the `e2e` testsuite is failing because of a crash in `ConcatModel`

**Comment by caybro** [comment]
> NB: the `e2e` testsuite is failing because of a crash in `ConcatModel`

Fixed now

File: ui/StatusQ/src/StatusQ/Controls/StatusBaseInput.qml (hunk 4 above)
[the remark this answers is not in the document]
Comment (alexjba) [reply ui/StatusQ/src/StatusQ/Controls/StatusBaseInput.qml:281]: IMO it looks much better in dark mode. As opposed to `Theme.palette.baseColor2`

Screenshot 2023-12-21 at 15 19 58

[the remark this answers is not in the document]
Comment (alexjba) [reply ui/StatusQ/src/StatusQ/Controls/StatusBaseInput.qml:281]: > IMO it looks much better in dark mode. As opposed to `Theme.palette.baseColor2`
> 
> Screenshot 2023-12-21 at 15 19 58

But the colours on the section headers are wrong IMO. If on light theme it's the same colour as the delegate item, I guess it should be the same colour on the dark theme as well.

[the remark this answers is not in the document]
Comment (caybro) [reply ui/StatusQ/src/StatusQ/Controls/StatusBaseInput.qml:281]: Yeah it's really a different color based on the light/dark theme

**Comment by caybro** [comment]
> > Also another detail. The searcher placeholder should be:
> > Screenshot 2023-12-19 at 14 15 36
> > And allow the search by name or id. [Design](https://www.figma.com/file/eM26pyHZUeAwMLviaS1KJn/%E2%9A%99%EF%B8%8F-Wallet-Settings%3A-Manage-Tokens?type=design&node-id=71%3A103480&mode=design&t=HBhmgzvmu47bpf8o-1)
> 
> Right, I'll extend the search to include the collection ID as well

FTR, this has been implemented as well; the search now also include the ID

**Review by alexjba — COMMENTED** [review]
I'll leave this partial review and continue with the code.

I was wondering why the checkboxes are grey in the design. But I think I got it now. The default state should be with all the checkboxes checked and marked by using a grey background (see the screenshot below). When the first checkbox gets checked by the user the colour changes to blue and all the other checkboxes are unchecked. We're missing this flow.

Screenshot 2023-12-21 at 14 58 57

The header is flickering when an item is selected/deselected

https://github.com/status-im/status-desktop/assets/47811206/963048df-7344-4f71-9315-a24fae04c0f5

The middle section drops a shadow when the top section drops it. The shadow should be decoupled.

Screenshot 2023-12-21 at 15 24 02

Should the `Collection` button change its state while searching?

https://github.com/status-im/status-desktop/assets/47811206/1fe25f0c-c8bc-4372-b9d2-0f2fc367c6bd

**Comment by caybro** [comment]
> I'll leave this partial review and continue with the code.
> 
> I was wondering why the checkboxes are grey in the design. But I think I got it now. The default state should be with all the checkboxes checked and marked by using a grey background (see the screenshot below). When the first checkbox gets checked by the user the colour changes to blue and all the other checkboxes are unchecked. We're missing this flow.

Ah, only noticed now! OK, will have to redo the logic a bit there :/ Good catch!

> The header is flickering when an item is selected/deselected

Yup, will fix, probably the OpacityAnimation there
 
> The middle section drops a shadow when the top section drops it. The shadow should be decoupled.

Nice catch too, never noticed

> Should the `Collection` button change its state while searching?

Nope, it should only when the popup is closed, gonna fix

**Comment by alexjba** [comment]
> > I'll leave this partial review and continue with the code.
> > I was wondering why the checkboxes are grey in the design. But I think I got it now. The default state should be with all the checkboxes checked and marked by using a grey background (see the screenshot below). When the first checkbox gets checked by the user the colour changes to blue and all the other checkboxes are unchecked. We're missing this flow.
> 
> Ah, only noticed now! OK, will have to redo the logic a bit there :/ Good catch!

I think the `Clear filter` button needs to be considered as well in the new logic. Once everything is checked it's kind of redundant because it will uncheck everything, but the filtering doesn't change.

**Comment by caybro** [comment]
> > > I'll leave this partial review and continue with the code.
> > > I was wondering why the checkboxes are grey in the design. But I think I got it now. The default state should be with all the checkboxes checked and marked by using a grey background (see the screenshot below). When the first checkbox gets checked by the user the colour changes to blue and all the other checkboxes are unchecked. We're missing this flow.
> > 
> > 
> > Ah, only noticed now! OK, will have to redo the logic a bit there :/ Good catch!
> 
> I think the `Clear filter` button needs to be considered as well in the new logic. Once everything is checked it's kind of redundant because it will uncheck everything, but the filtering doesn't change.

Yeah, that was my initial idea too ... but then I figured out it might be enough to just change the appearance of the checkbox itself, not the logic :)

**Comment by caybro** [comment]
> > > > I'll leave this partial review and continue with the code.
> > > > I was wondering why the checkboxes are grey in the design. But I think I got it now. The default state should be with all the checkboxes checked and marked by using a grey background (see the screenshot below). When the first checkbox gets checked by the user the colour changes to blue and all the other checkboxes are unchecked. We're missing this flow.
> > > 
> > > 
> > > Ah, only noticed now! OK, will have to redo the logic a bit there :/ Good catch!
> > 
> > 
> > I think the `Clear filter` button needs to be considered as well in the new logic. Once everything is checked it's kind of redundant because it will uncheck everything, but the filtering doesn't change.
> 
> Yeah, that was my initial idea too ... but then I figured out it might be enough to just change the appearance of the checkbox itself, not the logic :)

Fixed as well:
[Checkboxes](https://github.com/status-im/status-desktop/assets/5377645/66c87440-6ec1-4e91-a6fe-3b6e04883f2c)

**Review by alaibe — APPROVED** [review]

**Review by Khushboo-dev-cpp — APPROVED** [review]
Have just had a brief look at the PR and looks good to me.

**Pull request merged**